### PR TITLE
Add FormRecognizerAudience and DocumentAnalysisAudience to Azure.AI.FormRecognizer

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by nisha-bhatia** [comment]
Resolves: https://github.com/azure/azure-sdk-for-net/issues/17192

File: sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 - Added the `tags` parameter to the `StartCreateComposedModel` methods.
 
 ### Bugs Fixed
+- FormRecognizerAudience and DocumentAnalysisAudience has been added to allow the user to select the Azure cloud where the resource is located. Issue [17192](https://github.com/azure/azure-sdk-for-net/issues/17192).
```
Comment (jsquire) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md:15]:
```suggestion
- FormRecognizerAudience and DocumentAnalysisAudience have been added to allow the user to select the Azure cloud where the resource is located. Issue [17192](https://github.com/azure/azure-sdk-for-net/issues/17192).
```

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisAudience.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    /// <summary> Cloud audiences available for FormRecognizer.DocumentAnalysis. </summary>
+    public readonly partial struct DocumentAnalysisAudience : IEquatable<DocumentAnalysisAudience>
+    {
+        private readonly string _value;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DocumentAnalysisAudience"/> object.
+        /// </summary>
+        /// <param name="value">The Azure Active Directory audience to use when forming authorization scopes. For FormRecognizer.DocumentAnalysis, this value corresponds to a URL that identifies the Azure cloud where the resource is located. For more information: <see href="https://docs.microsoft.com/azure/azure-government/documentation-government-cognitiveservices" />.</param>
+        /// <exception cref="ArgumentNullException"> <paramref name="value"/> is null. </exception>
```
Comment (jsquire) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisAudience.cs:19]: Should we add line for `ArgumentException` when `value` is empty?   (that's the behavior of `AssertNotNullOrEmpty`)

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientOptions.cs [hunk 3]
```diff
@@ -28,6 +28,9 @@ public DocumentAnalysisClientOptions(ServiceVersion version = LatestVersion)
             };
 
             AddLoggedHeadersAndQueryParameters();
+
+            //Default Audience to Azure Public Cloud
+            Audience ??= DocumentAnalysisAudience.AzurePublicCloud;
```
Comment (jsquire) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClientOptions.cs:33]:
```suggestion
```

We want the options to default to `null`; the client will apply the default.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClient/FormRecognizerClientOptions.cs [hunk 4]
```diff
@@ -24,6 +30,9 @@ public FormRecognizerClientOptions(ServiceVersion version = LatestVersion)
         {
             Version = version;
             AddLoggedHeadersAndQueryParameters();
+
+            //Default Audience to Azure Public Cloud
+            Audience ??= FormRecognizerAudience.AzurePublicCloud;
```
Comment (jsquire) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClient/FormRecognizerClientOptions.cs:35]:
```suggestion
```

We want the options to default to `null`; the client is applying the default.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClient/FormRecognizerAudience.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    /// <summary> Cloud audiences available for FormRecognizer. </summary>
+    public readonly partial struct FormRecognizerAudience : IEquatable<FormRecognizerAudience>
+    {
+        private readonly string _value;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FormRecognizerAudience"/> object.
+        /// </summary>
+        /// <param name="value">The Azure Active Directory audience to use when forming authorization scopes. For FormRecognizer, this value corresponds to a URL that identifies the Azure cloud where the resource is located. For more information: <see href="https://docs.microsoft.com/azure/azure-government/documentation-government-cognitiveservices" />.</param>
+        /// <exception cref="ArgumentNullException"> <paramref name="value"/> is null. </exception>
```
Comment (jsquire) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClient/FormRecognizerAudience.cs:19]: Should we add line for `ArgumentException` when `value` is empty?   (that's the behavior of `AssertNotNullOrEmpty`)

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClient.cs [hunk 6]
```diff
@@ -97,8 +97,10 @@ public DocumentAnalysisClient(Uri endpoint, TokenCredential credential, Document
 
             options ??= new DocumentAnalysisClientOptions();
 
+            string defaultScope = $"{(string.IsNullOrEmpty(options.Audience.ToString()) ? DocumentAnalysisAudience.AzurePublicCloud : options.Audience)}/.default";
```
Comment (jsquire) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentAnalysisClient.cs:100]:
```suggestion
            string defaultScope = $"{(string.IsNullOrEmpty(options.Audience?.ToString()) ? DocumentAnalysisAudience.AzurePublicCloud : options.Audience)}/.default";
```

We need to account for the audience being `null`.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentModelAdministrationClient.cs [hunk 7]
```diff
@@ -103,9 +103,10 @@ public DocumentModelAdministrationClient(Uri endpoint, TokenCredential credentia
             Argument.AssertNotNull(credential, nameof(credential));
 
             options ??= new DocumentAnalysisClientOptions();
+            string defaultScope = $"{(string.IsNullOrEmpty(options.Audience.ToString()) ? DocumentAnalysisAudience.AzurePublicCloud : options.Audience)}/.default";
```
Comment (jsquire) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentModelAdministrationClient.cs:106]:
```suggestion
            string defaultScope = $"{(string.IsNullOrEmpty(options.Audience?.ToString()) ? DocumentAnalysisAudience.AzurePublicCloud : options.Audience)}/.default";
```

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient/FormTrainingClient.cs [hunk 8]
```diff
@@ -107,9 +107,11 @@ public FormTrainingClient(Uri endpoint, TokenCredential credential, FormRecogniz
             Argument.AssertNotNull(credential, nameof(credential));
             Argument.AssertNotNull(options, nameof(options));
 
+            string defaultScope = $"{(string.IsNullOrEmpty(options.Audience.ToString()) ? FormRecognizerAudience.AzurePublicCloud : options.Audience)}/.default";
```
Comment (jsquire) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient/FormTrainingClient.cs:110]:
```suggestion
            string defaultScope = $"{(string.IsNullOrEmpty(options.Audience?.ToString()) ? FormRecognizerAudience.AzurePublicCloud : options.Audience)}/.default";
```

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClient/FormRecognizerClient.cs [hunk 9]
```diff
@@ -96,8 +96,10 @@ public FormRecognizerClient(Uri endpoint, TokenCredential credential, FormRecogn
             Argument.AssertNotNull(credential, nameof(credential));
             Argument.AssertNotNull(options, nameof(options));
 
+            string defaultScope = $"{(string.IsNullOrEmpty(options.Audience.ToString()) ? FormRecognizerAudience.AzurePublicCloud : options.Audience)}/.default";
```
Comment (jsquire) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClient/FormRecognizerClient.cs:99]:
```suggestion
            string defaultScope = $"{(string.IsNullOrEmpty(options.Audience?.ToString()) ? FormRecognizerAudience.AzurePublicCloud : options.Audience)}/.default";
```

**Review by jsquire — APPROVED** [review]
LGTM

**Pull request merged**